### PR TITLE
Use newer ECMAScript code

### PR DESCRIPTION
**Pull request by DomAmato**
Currently there is some inconsistent use of things introduced from ES6 onward and while this is not a comprehensive change of all elements ES6 introduced it does convert the following:

- use default parameters where possible
- the exponentiation operator `**` over `math.pow`
- uses `includes` instead of `~indexOf() || ~indexOf() === -1`
- use property shorthand where possible
- converted function expressions to arrow expressions

### CODE REVIEW

File: src/controllers/things_controller.js [hunk 1]
```diff
@@ -566,7 +566,7 @@ ThingsController.ws('/:thingId/', (websocket, request) => {
           data: {
             status: '400 Bad Request',
             message: `Unknown messageType: ${request.messageType}`,
-            request: request,
+            rrequest,
```
Comment (mrstegeman) [inline src/controllers/things_controller.js]: typo: `request`

**Comment by DomAmato** [comment]
Thanks I will run that, I didn't see a style guide but whatever the linter returns I'll fix and commit it :)

**Comment by mrstegeman** [comment]
`eslint` still find lots of issues:

```
/home/travis/build/mozilla-iot/gateway/src/models/settings.js
  40:1   error  Expected indentation of 18 spaces but found 8  indent
  40:29  error  There should be no space after '{'             object-curly-spacing
  40:53  error  There should be no space before '}'            object-curly-spacing
  45:1   error  Expected indentation of 18 spaces but found 6  indent
  45:27  error  There should be no space after '{'             object-curly-spacing
  45:51  error  There should be no space before '}'            object-curly-spacing
  80:1   error  Expected indentation of 18 spaces but found 8  indent
/home/travis/build/mozilla-iot/gateway/src/models/user.js
  15:22  error  Unexpected space before function parentheses  space-before-function-paren
  32:41  error  Unexpected space before function parentheses  space-before-function-paren
/home/travis/build/mozilla-iot/gateway/src/models/users.js
  23:1  error  Expected indentation of 4 spaces but found 6  indent
  24:1  error  Expected indentation of 6 spaces but found 8  indent
  25:1  error  Expected indentation of 4 spaces but found 6  indent
  26:1  error  Expected indentation of 4 spaces but found 6  indent
  27:1  error  Expected indentation of 2 spaces but found 4  indent
  58:1  error  Expected indentation of 4 spaces but found 6  indent
  59:1  error  Expected indentation of 6 spaces but found 8  indent
  60:1  error  Expected indentation of 4 spaces but found 6  indent
  61:1  error  Expected indentation of 2 spaces but found 4  indent
/home/travis/build/mozilla-iot/gateway/src/passwords.js
  41:5  error  Expected no linebreak before this expression  implicit-arrow-linebreak
  50:5  error  Expected no linebreak before this expression  implicit-arrow-linebreak
/home/travis/build/mozilla-iot/gateway/src/router-setup.js
   15:3   error  Expected no linebreak before this expression  implicit-arrow-linebreak
   30:31  error  There should be no space after '{'            object-curly-spacing
   30:49  error  There should be no space before '}'           object-curly-spacing
  124:37  error  There should be no space after '{'            object-curly-spacing
  124:44  error  There should be no space before '}'           object-curly-spacing
  202:47  error  There should be no space after '{'            object-curly-spacing
  202:62  error  There should be no space before '}'           object-curly-spacing
/home/travis/build/mozilla-iot/gateway/src/rules-engine/Database.js
  35:39  error  Unexpected space before function parentheses   space-before-function-paren
  92:1   error  Line 92 exceeds the maximum line length of 80  max-len
/home/travis/build/mozilla-iot/gateway/src/ssltunnel.js
   53:1   error  Expected indentation of 35 spaces but found 8   indent
   55:1   error  Expected indentation of 32 spaces but found 10  indent
   65:1   error  Expected indentation of 22 spaces but found 8   indent
   65:9   error  There should be no space after '{'              object-curly-spacing
   65:50  error  There should be no space before '}'             object-curly-spacing
   70:28  error  Unexpected space before function parentheses    space-before-function-paren
   75:18  error  Unexpected space before function parentheses    space-before-function-paren
   84:1   error  Expected indentation of 20 spaces but found 12  indent
   85:1   error  Expected indentation of 22 spaces but found 14  indent
   86:1   error  Expected indentation of 21 spaces but found 14  indent
   87:1   error  Expected indentation of 21 spaces but found 14  indent
   88:1   error  Expected indentation of 23 spaces but found 14  indent
   89:1   error  Expected indentation of 23 spaces but found 14  indent
   90:1   error  Expected indentation of 20 spaces but found 12  indent
   90:13  error  There should be no space after '{'              object-curly-spacing
   90:27  error  There should be no space before '}'             object-curly-spacing
  152:17  error  Unexpected space before function parentheses    space-before-function-paren
  168:5   error  Expected no linebreak before this expression    implicit-arrow-linebreak
/home/travis/build/mozilla-iot/gateway/src/user-profile.js
  25:17  error  Unexpected space before function parentheses  space-before-function-paren
  74:20  error  Unexpected space before function parentheses  space-before-function-paren
/home/travis/build/mozilla-iot/gateway/static/js/schema-form/schema-utils.js
   52:13  error  There should be no space after '{'              object-curly-spacing
   52:36  error  There should be no space before '}'             object-curly-spacing
   54:41  error  There should be no space after '{'              object-curly-spacing
   54:73  error  There should be no space before '}'             object-curly-spacing
   55:1   error  Expected indentation of 40 spaces but found 8   indent
   56:1   error  Expected indentation of 40 spaces but found 8   indent
   59:1   error  Expected indentation of 61 spaces but found 8   indent
   60:1   error  Expected indentation of 61 spaces but found 8   indent
   62:1   error  Expected indentation of 40 spaces but found 8   indent
   63:1   error  Expected indentation of 40 spaces but found 8   indent
   71:11  error  There should be no space after '{'              object-curly-spacing
   71:47  error  There should be no space before '}'             object-curly-spacing
   83:1   error  Expected indentation of 61 spaces but found 10  indent
   86:1   error  Expected indentation of 57 spaces but found 10  indent
   87:1   error  Expected indentation of 57 spaces but found 10  indent
   88:1   error  Expected indentation of 57 spaces but found 10  indent
   89:1   error  Expected indentation of 57 spaces but found 10  indent
  103:12  error  There should be no space after '{'              object-curly-spacing
  103:34  error  There should be no space before '}'             object-curly-spacing
  113:11  error  There should be no space after '{'              object-curly-spacing
  113:39  error  There should be no space before '}'             object-curly-spacing
  122:1   error  Expected indentation of 42 spaces but found 8   indent
  123:1   error  Expected indentation of 42 spaces but found 8   indent
  124:1   error  Expected indentation of 42 spaces but found 8   indent
  125:1   error  Expected indentation of 42 spaces but found 8   indent
  144:13  error  There should be no space after '{'              object-curly-spacing
  144:56  error  There should be no space before '}'             object-curly-spacing
  152:15  error  There should be no space after '{'              object-curly-spacing
  152:24  error  There should be no space before '}'             object-curly-spacing
  153:1   error  Expected indentation of 54 spaces but found 10  indent
  173:29  error  There should be no space after '{'              object-curly-spacing
  173:76  error  There should be no space before '}'             object-curly-spacing
  180:1   error  Line 180 exceeds the maximum line length of 80  max-len
  231:1   error  Expected indentation of 49 spaces but found 8   indent
  232:1   error  Expected indentation of 49 spaces but found 8   indent
  245:1   error  Expected indentation of 46 spaces but found 8   indent
  246:1   error  Expected indentation of 46 spaces but found 8   indent
  247:1   error  Expected indentation of 46 spaces but found 8   indent
  276:16  error  There should be no space after '{'              object-curly-spacing
  276:31  error  There should be no space before '}'             object-curly-spacing
  283:16  error  There should be no space after '{'              object-curly-spacing
  283:31  error  There should be no space before '}'             object-curly-spacing
  406:16  error  There should be no space after '{'              object-curly-spacing
  406:31  error  There should be no space before '}'             object-curly-spacing
  413:16  error  There should be no space after '{'              object-curly-spacing
  413:31  error  There should be no space before '}'             object-curly-spacing
```

**Comment by DomAmato** [comment]
Sorry I was trying to configure our tool to do as much of the fixes as possible but the formatting is still something we are trying to fix. No more linter complaints

**Comment by mrstegeman** [comment]
@DomAmato There are 36 failing tests now: https://travis-ci.org/mozilla-iot/gateway/jobs/503220190

**Comment by DomAmato** [comment]
looks like its likely due to the scope change with `this` when using arrow functions vs the previous method. 
https://github.com/kale-code/gateway/commit/b9e8c46d1c884c458c3bb71440bf0e2e4e122574#diff-12557e6619cc8236e461c9c2865ea4e0L114

I can revert it for now.

**Comment by mrstegeman** [comment]
Thanks for fixing things up.

@hobinjk Do you have any objections to this?

**Comment by hobinjk** [comment]
@mrstegeman No, I did a few spot-checks and everything looked reasonable. I'm impressed at the breadth of cases kale was able to handle correctly.

**Pull request merged**